src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceMetadataBehavior.cs

```diff
@@ -162,6 +162,14 @@ private void SetExtensionProperties(ServiceDescription description, ServiceHostB
             }
         }
 
+        private static Uri GetFirstEndpointUriForScheme(string uriScheme, Uri configuredGetUrl, ServiceDescription description)
+            => configuredGetUrl
+                ?? description?.Endpoints?.FirstOrDefault(endpoint =>
+                       endpoint?.Address?.Uri != null
+                       && endpoint.Address.Uri.Scheme == uriScheme)
```

Comment:
I think you can simplify this:
```suggestion
                       endpoint?.Address?.Uri?.Scheme == uriScheme)
```

Reply:
In the past I had different experiences with something that ended up as

`(string)null == string `

which is then actually executed as 

`(string)null.Equals(string)`

wich then yielded a NullReferenceException. But I'm not sure wich language level is targeted in this project currenty, this behaviour seemed to have changed in the last years between the different framework versions.

My preference is

`string.Equals(endpoint?.Address?.Uri?.Scheme, uriScheme) `

but that is only 100% correct when uriScheme is never given as null, therefore I wrote the separate explicit null-check.

Reply:
I usually like string.Equals too as you can specify the string comparison mode and skip some null checks. FYI `(string)null == string` will use the string == operator method, which calls into the static string.Equals method. One advantage of using string.Equals over == is you can't accidentally call the object == operation. If you have an object you expect to be a string and forget to cast to string first, == could do something different than you intend. Feel free to switch to string.Equals if you prefer, but use the overload which takes a StringComparison and pass in StringComparison.Ordinal.